File: 0900-reordered-power-of-2/0900-reordered-power-of-2.java
class Solution {
    public boolean reorderedPowerOf2(int n) {
        String target = sortDigits(n);

        Set<String> powerSet = new HashSet<>();
        for(int i = 0; i < 31; i++){
            int pow = 1 << i;
            powerSet.add(sortDigits(pow));
        }

        return powerSet.contains(target);
        
    }

    private String sortDigits(int num){
        char[] arr = String.valueOf(num).toCharArray();
        Arrays.sort(arr);
        return new String(arr);
    }
}

File: 0900-reordered-power-of-2/0900-reordered-power-of-2.py
class Solution:
    def reorderedPowerOf2(self, n: int) -> bool:
        def sort_value(x):
            return ''.join(sorted(str(x)))

        power_set = {sort_value(1 << i) for i in range(31)}

        return sort_value(n) in power_set


        